src/hotspot/share/services/diagnosticCommand.cpp

```diff
@@ -1045,16 +1047,15 @@ void CheckpointDCmd::execute(DCmdSource source, TRAPS) {
   JavaValue result(T_OBJECT);
   JavaCallArguments args;
   args.push_long((jlong)output());
-  LinuxAttachOperation* current_op;
   JavaCalls::call_static(&result, k,
                          vmSymbols::checkpointRestoreInternal_name(),
                          vmSymbols::checkpointRestoreInternal_signature(), &args, CHECK);
   oop str = result.get_oop();
   if (str != NULL) {
     char* out = java_lang_String::as_utf8_string(str);
     if (out[0] != '\0') {
-      current_op = LinuxAttachListener::get_current_op();
-      outputStream* stream = current_op->is_effectively_completed() ? tty : output();
+      outputStream* stream = output();
+      LINUX_ONLY(if (LinuxAttachListener::get_current_op() && LinuxAttachListener::get_current_op()->is_effectively_completed()) { stream = tty; })
```

Comment:
A multiline comment would look better
```
#ifdef LINUX
...
```

The original code did not check LinuxAttachListener::get_current_op() == NULL, as at this point the operation should exist. I propose omit the check or replace that with assert() at most.